### PR TITLE
BuildEngineApiClient

### PR DESCRIPTION
**Pull request by chrisvire**
Client Wrapper for BuildEngineAPI
- [x] Accept BaseUrl and ApiAccessKey
- [x] expose System Check
- [x] expose Project Create/Get/GetAll/Delete (no Update)

### CODE REVIEW

File: source/SIL.AppBuilder.BuildEngineApiClient.Tests/Integration/ProjectTests.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+namespace SIL.AppBuilder.BuildEngineApiClient.Tests.Integration
+{
+    public class ProjectTests
+    {
+        // Note: You should change these values to match your environment.  These are not 
+        public string BaseUrl { get; set; } = "https://buildengine.gtis.guru";
+        public string ApiAccessKey { get; set; } = "BNLAVlQ7xlBoSqvRNbS153FqCWfcVCpX";
+        public string UserId { get; set; } = "chris_hubbard@sil.org";
+        public string GroupId { get; set; } = "LSDEV";
+        public string PublishingKey { get; set; } = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDik3IL0ldYovV6VYX3ICczNwJS2fl2fHkfaki/P4PWyl0p1glg/K5V74Vun9S3Fv+AwrMu8vUVXSuinINYqeFzIwc7M5r0QxAy60ExBZQiXYPs//maTJdyXkwZQkIQQNTf+NT6LBhNqv6iekzDO2AL/cirVq2wbuO8bzlBLLUc9uwLcXVd7IGZyJ9LLB5/T7gRPQAGif85m7lRAJ1R2+2QnPsBTMjyqcrKULSysLqd5+xXChljxSiJqVjGAh0jqpD+W7vo85iC9wqmq0H+MYTzE86qoXMh12k/boDIeZHBnkjRAZyRgq5uDvyVFk0c6Hn7fAGWvg04sWDr77yGNpIl hubbard@swd-hubbard-mac.local";
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.BuildEngineApiClient.Tests/Integration/ProjectTests.cs]: did you intend to check this in? 

do you need to revoke access to things now? 

:astonished:

Comment (chrisvire) [reply source/SIL.AppBuilder.BuildEngineApiClient.Tests/Integration/ProjectTests.cs]: It is a public key ... it is an example.  I didn't finish the comment ... sigh...

File: source/SIL.AppBuilder.BuildEngineApiClient.Tests/Integration/SystemTests.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using Xunit;
+namespace SIL.AppBuilder.BuildEngineApiClient.Tests.Integration
+{
+    public class SystemTests
+    {
+        public string BaseUrl { get; set; } = "https://buildengine.gtis.guru";
+        public string ApiAccessKey { get; set; } = "BNLAVlQ7xlBoSqvRNbS153FqCWfcVCpX";
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.BuildEngineApiClient.Tests/Integration/SystemTests.cs]: is this a valid key?

File: source/SIL.AppBuilder.BuildEngineApiClient/SnakeJsonSerializerStrategy.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Linq;
+using SimpleJson;
+
+namespace SIL.AppBuilder.BuildEngineApiClient
+{
+    public class SnakeJsonSerializerStrategy : PocoJsonSerializerStrategy
+    {
+        protected override string MapClrMemberNameToJsonFieldName(string clrPropertyName)
+        {
+            //PascalCase to snake_case
```
[the remark this answers is not in the document]
Comment (chrisvire) [reply source/SIL.AppBuilder.BuildEngineApiClient/SnakeJsonSerializerStrategy.cs]: @NullVoxPopuli, Newtonsoft.JSON has support for snake case, but RestSharp did work to remove the dependency.  I am not sure if it is worth adding a nuget dependency for this short amount of code.
I did followed the [documentation recommendation](https://github.com/restsharp/RestSharp/wiki/Deserialization#overriding-jsonserializationstrategy) for overriding the serialization strategy.  I found a bug in their sample and [reported it](https://github.com/restsharp/RestSharp/issues/1159).

[the remark this answers is not in the document]
Comment (chrisvire) [reply source/SIL.AppBuilder.BuildEngineApiClient/SnakeJsonSerializerStrategy.cs]: It is a one-liner overriding a strategy class which is a documented extensibility point in RestSharp.  I am not sure why I would want to add a package dependency for that.

[the remark this answers is not in the document]
Comment (chrisvire) [reply source/SIL.AppBuilder.BuildEngineApiClient/SnakeJsonSerializerStrategy.cs]: OK. I will use Humanizer.

**Comment by chrisvire** [comment]
@NullVoxPopuli, so what is left to accept this PR?

**Pull request merged**